Syndics honor MoM acl

dwoz opened the pull request:
### What does this PR do?

This is an alternate approach to #63257

### What issues does this PR fix or reference?
Fixes: 

#62618
#62577

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

dwoz committed:
Syndics honor MoM acl

dwoz committed:
Auth errors result in error events

waynew commented on tests/pytests/unit/utils/test_minions.py:
```diff
@@ -65,7 +65,7 @@ def test_validate_tgt_should_return_false_when_no_valid_minions_have_been_found(
         "salt.utils.minions.CkMinions.check_minions", autospec=True, return_value={}
     ):
         result = ckminions.validate_tgt("fnord", "fnord", "fnord", minions=[])
-        assert result is False
```
FWIW this test should either be renamed or completely removed - this was part of the CVE check to ensure that we weren't validating minions that weren't valid.

dwoz replied:
Agreed, I've renamed the test and also added a doc string to help explain what's going on.

waynew reviewed (changes requested):
Only one problem that I see, and that's in the test suite.

I'm going to check this out and bang on it a bit more. I'm also seeing some failures here: https://github.com/saltstack/salt/actions/runs/3864153908/jobs/6586792092 that are concerning

dwoz committed:
Address test issues

dwoz committed:
Re-use request clients

dwoz committed:
Update docs

dwoz committed:
Add changelogs

waynew committed:
Update tests

We want to ensure that targeting invalid minions does *not* allow us to
run any sort of shenanigans on the minions in question! This ensures
that valid commands (i.e. file.touch in this case) cannot inadvertently
be sent to minions that we don't have ACL for.

waynew reviewed (changes requested):
Okay, I opened a PR against your fork and I've also added I think what would make the docstring :+1: - if you want to finish this up I can close mine in favor of this PR, or vice versa and I can go ahead and combine these changes with my PR.

Those smol changes and this is ready to rock! :tada:

Ch3LL commented on doc/topics/hardening.rst:
```diff
@@ -150,3 +150,13 @@ Run the following on the Salt minion:
+==========================
+
+Syndics must be run as the same user as their syndic master process. The master
+of master's will include publisher ACL information in jobs send to downstream
```
```suggestion
of master's will include publisher ACL information in jobs sent to downstream
```

Ch3LL commented:
Looks like the test failures are possibly related to this change.

dwoz committed:
Fix typo

dwoz committed:
Test purging of docker resources before tests run

dwoz committed:
Remove docker puge logic

dwoz committed:
Pam eauth not available on windows

dwoz committed:
Fix isort precommit

waynew committed:
Use test instead of ls

It's possible that the output of ls is different for some reason, but
test -f should always return 0 if the file exists, or 1 otherwise. We
want it to be not 0.

waynew committed:
Workaround for docker images missing dependencies
